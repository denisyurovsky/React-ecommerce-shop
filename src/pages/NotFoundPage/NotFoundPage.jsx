import React from 'react';
import { Link } from 'react-router-dom';

import styles from './NotFoundPage.module.scss';

export function NotFoundPage() {
  return (
    <div className={styles.container}>
      <p className={styles.title}>Page Not Found</p>
      <p>The page you are looking for could not be found.</p>
      <Link to={'/'}>Go to home page</Link>
    </div>
  );
}
