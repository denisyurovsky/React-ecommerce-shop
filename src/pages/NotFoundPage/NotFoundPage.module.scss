.container {
  margin: 0 auto;
  text-align: center;
  font-size: 24px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}

.title {
  font-size: 48px;
}
