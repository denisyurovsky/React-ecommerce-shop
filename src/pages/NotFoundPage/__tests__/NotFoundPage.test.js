import { render, screen } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import { NotFoundPage } from '../NotFoundPage';

describe('NotFoundPage component', () => {
  beforeEach(() => {
    render(<NotFoundPage />, { wrapper: MemoryRouter });
  });

  it('renders a valid snapshot', () => {
    const { asFragment } = render(<NotFoundPage />, {
      wrapper: MemoryRouter,
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders "Go to home page" as a link to app home page', () => {
    const courseLink = screen.getByText(/home page/i);

    expect(courseLink.getAttribute('href')).toBe('/');
  });
});
