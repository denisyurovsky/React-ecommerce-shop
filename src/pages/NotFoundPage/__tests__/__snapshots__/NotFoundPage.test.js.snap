// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p
      class="title"
    >
      Page Not Found
    </p>
    <p>
      The page you are looking for could not be found.
    </p>
    <a
      href="/"
    >
      Go to home page
    </a>
  </div>
</DocumentFragment>
`;
